sov-modules/sov-state/src/prover_storage.rs

```diff
@@ -101,6 +101,7 @@ impl<S: MerkleProofSpec> Storage for ProverStorage<S> {
             let key_hash = KeyHash(S::Hasher::hash(key.key.as_ref()));
             // TODO: Switch to the batch read API once it becomes available
             let (result, proof) = untracked_jmt.get_with_proof(key_hash, latest_version)?;
+            // TODO: Question, wouldn't witness be "polluted" in case of error from below?
```

Comment:
This

Reply:
Hmm, yes I think it would. So maybe we should use `assert` instead of `ensure!`

Reply:
assert will panic. Fixed it with differnt approach